test/parallel/test-process-env-allowed-flags-are-documented.js

```diff
@@ -53,15 +53,6 @@ const conditionalOpts = [
         '--secure-heap-min',
       ].includes(opt);
     }
-  }, {
-    // We are using openssl_is_fips from the configuration because it could be
-    // the case that OpenSSL is FIPS compatible but fips has not been enabled
-    // (starting node with --enable-fips). If we use common.hasFipsCrypto
-    // that would only tells us if fips has been enabled, but in this case we
-    // want to check options which will be available regardless of whether fips
-    // is enabled at runtime or not.
-    include: process.config.variables.openssl_is_fips,
-    filter: (opt) => opt.includes('-fips')
```

Comment:
We also need to test when node is configured `without-ssl`:
```console
./configure --without-ssl && make -j8
```
With that configuration this test and `test/parallel/test-cli-node-print-help.js` are currently [failing](https://ci.nodejs.org/job/node-test-commit-linux-containered/nodes=ubuntu1804_sharedlibs_withoutssl_x64/25198/).

`test-cli-node-print-help.js` can be run using the following command:
```console
$ ./node --expose-internals test/parallel/test-cli-node-print-help.j
```

Reply:
I have moved the FIPS flags behind the same checks as the rest of the SSL-related flags.

Thanks for pointing the `--without-ssl` compilation out!